src/platform/forms-system/src/js/components/FormNav.jsx

```diff
@@ -11,82 +9,96 @@ import {
-    // finding the current page, then getting the chapter name using the key
-    const formPages = createFormPageList(formConfig);
-    const pageList = createPageList(formConfig, formPages);
+  const eligiblePageList = getActiveExpandedPages(pageList, formData);
 
-    const eligiblePageList = getActiveExpandedPages(pageList, formData);
+  const chapters = _.uniq(
+    eligiblePageList.map(p => p.chapterKey).filter(key => !!key),
+  );
 
-    const chapters = _.uniq(
-      eligiblePageList.map(p => p.chapterKey).filter(key => !!key),
+  let page = eligiblePageList.filter(p => p.path === currentPath)[0];
+  // If the page isn’t active, it won’t be in the eligiblePageList
+  // This is a fallback to still find the chapter name if you open the page directly
+  // (the chapter index will probably be wrong, but this isn’t a scenario that happens in normal use)
+  if (!page) {
+    page = formPages.find(
+      p => `${formConfig.urlPrefix}${p.path}` === currentPath,
     );
+  }
 
-    let page = eligiblePageList.filter(p => p.path === currentPath)[0];
-    // If the page isn’t active, it won’t be in the eligiblePageList
-    // This is a fallback to still find the chapter name if you open the page directly
-    // (the chapter index will probably be wrong, but this isn’t a scenario that happens in normal use)
-    if (!page) {
-      page = formPages.find(
-        p => `${formConfig.urlPrefix}${p.path}` === currentPath,
-      );
+  let current;
+  let chapterName;
+  if (page) {
+    current = chapters.indexOf(page.chapterKey) + 1;
+    // The review page is always part of our forms, but isn’t listed in chapter list
+    chapterName =
+      page.chapterKey === 'review'
+        ? formConfig?.customText?.reviewPageTitle || REVIEW_APP_DEFAULT_MESSAGE
+        : formConfig.chapters[page.chapterKey].title;
+    if (typeof chapterName === 'function') {
+      chapterName = chapterName();
     }
+  }
+
+  const stepText = `Step ${current} of ${chapters.length}: ${chapterName}`;
+  const showHeader = Math.abs(current - index) === 1;
 
-    let current;
-    let chapterName;
-    if (page) {
-      current = chapters.indexOf(page.chapterKey) + 1;
-      // The review page is always part of our forms, but isn’t listed in chapter list
-      chapterName =
-        page.chapterKey === 'review'
-          ? formConfig?.customText?.reviewPageTitle ||
-            REVIEW_APP_DEFAULT_MESSAGE
-          : formConfig.chapters[page.chapterKey].title;
-      if (typeof chapterName === 'function') {
-        chapterName = chapterName();
+  // The goal with this is to quickly "remove" the header from the DOM, and
+  // immediately re-render the component with the header included. This should
+  // ensure that VoiceOver on iOS will pick up on the new <h2>
+  useEffect(
+    () => {
+      if (current > index + 1) {
+        setIndex(index + 1);
+      } else if (current === index) {
+        setIndex(index - 1);
       }
```

Comment:
This is a little confusing. Can you add a little clarity in the comments around how this works? I think calling out that `current` is changed when the chapter changes, and that when the chapter changes, we need to remove the heading element and re-display it so iOS will announce the new element's text. Bonus points for linking to the ticket.